.github: Set commit status to error when workflow are cancelled

pchaigno opened the pull request:
GitHub jobs are usually set to status 'error' when cancelled. We should do the same for ci-xxx jobs when they are cancelled. Having the state appear as an error clarifies that the author, janitor, and reviewers should take notice of that workflow's result.

nbusseneau reviewed (changes requested):
> it can also happen for other automatic reasons (not yet determined)

Did you see this happen? The only case I've seen workflows get cancelled is when they hit the timeout, which is expected.

I personally still think both user cancellation and timeout cancellation resulting in `pending` is better than marking the job as failed, because when a job is cancelled it should simply be retriggered. Keeping it marked as pending makes it clearer IMO.

In any case, if we do want to move forward with marking jobs as failure, then I would recommend we instead just refactor both in the existing failure state by replacing:

```yaml
- name: Set commit status to failure
  if: ${{ failure() }}
```

with

```yaml
- name: Set commit status to failure
  if: ${{ !success() }}
```

(or `failure() || cancelled()`)

pchaigno committed:
.github: Set commit status to error when workflow are cancelled

GitHub jobs are usually set to status 'error' when cancelled. We should
do the same for ci-xxx jobs when they are cancelled. Having the state
appear as an error clarifies that the author, janitor, and reviewers
should take notice of that workflow's result.

Signed-off-by: Paul Chaignon <paul@cilium.io>
